core-build/gulp-core-build-typescript/src/RSCTask.ts

```diff
@@ -6,62 +6,46 @@ import * as resolve from 'resolve';
 
 import {
   JsonFile,
-  IPackageJson
+  IPackageJson,
+  FileSystem,
+  PackageJsonLookup
 } from '@microsoft/node-core-library';
 import { GulpTask } from '@microsoft/gulp-core-build';
 import * as RushStackCompiler from '@microsoft/rush-stack-compiler';
 import { GCBTerminalProvider } from './GCBTerminalProvider';
 
 export interface IRSCTaskConfig extends Object {
   buildDirectory: string;
+}
 
-  /**
-   * This is the name of the rush-stack-compiler package, or the name of the package that
-   * extends the configuration of rush-stack-compiler. This defaults to "\@microsoft/rush-stack-compiler"
-   */
-  rushStackCompilerPackageName?: string;
+interface ITsconfig {
+  extends?: string;
 }
 
 export abstract class RSCTask<TTaskConfig extends IRSCTaskConfig> extends GulpTask<TTaskConfig> {
+  private static _rushStackCompilerPackagePathCache: { [buildFolder: string]: string } = {};
+  private static __packageJsonLookup: PackageJsonLookup | undefined; // tslint:disable-line:variable-name
+  private static get _packageJsonLookup(): PackageJsonLookup {
+    if (!RSCTask.__packageJsonLookup) {
+      RSCTask.__packageJsonLookup = new PackageJsonLookup();
+    }
+
+    return RSCTask.__packageJsonLookup;
+  }
+
   protected _terminalProvider: GCBTerminalProvider = new GCBTerminalProvider(this);
 
   protected _rushStackCompiler: typeof RushStackCompiler;
 
-  private __rushStackCompilerPackagePath: string | undefined; // tslint:disable-line:variable-name
   private get _rushStackCompilerPackagePath(): string {
-    if (!this.__rushStackCompilerPackagePath) {
-      try {
-        this.__rushStackCompilerPackagePath = resolve.sync(
-          this.taskConfig.rushStackCompilerPackageName!,
-          {
-            basedir: this.buildConfig.rootPath,
-            packageFilter: (pkg: IPackageJson) => {
-              pkg.main = 'package.json';
-              return pkg;
-            }
-          }
-        );
-
-        if (!this.__rushStackCompilerPackagePath) {
-          throw new Error();
-        }
-
-        this.__rushStackCompilerPackagePath = path.dirname(this.__rushStackCompilerPackagePath);
-      } catch (e) {
-        throw new Error(`Unable to find "${this.taskConfig.rushStackCompilerPackageName}" package.`);
-      }
+    if (!RSCTask._rushStackCompilerPackagePathCache[this.buildFolder]) {
```

Comment:
>_rushStackCompilerPackagePathCache [](start = 17, length = 34)

Use a Map.  Otherwise if your buildFolder is a word like `constructor`, it will erroneously be found in the cache (because it will find `Object.prototype.constructor`).

This is a pervasive, habitual mistake among developers.  It really needs a tslint rule.